ingestion/ingestion/ingest.py

```diff
@@ -181,24 +181,35 @@ def validate_and_upload_data(
 def prune_bucket(bucket: storage.Bucket | FakeBucket):
     """Prune the bucket and waits until the bucket is empty by checking it periodically."""
 
-    bucket.lifecycle_rules = [{"action": {"type": "Delete"}, "condition": {"age": 0}}]
-    bucket.patch()
+    yesterday = datetime.now(timezone.utc) - timedelta(days=1)
+    prune_rule = {
+        "action": {"type": "Delete"},
+        "condition": {"createdBefore": yesterday.strftime("%Y-%m-%d")},
+    }
+
+    def is_prune_lifecycle(rule: dict) -> bool:
+        return (
+            rule["action"]["type"] == "Delete"
+            and "createdBefore" in rule["condition"]
+            and len(rule["condition"].keys()) == 0
+        )
 
-    try:
-        sleep_interval = 10
-        while True:
-            has_blobs = len(list(bucket.list_blobs(max_results=1))) != 0
-            if not has_blobs:
-                break
-
-            logger.info(f"bucket '{bucket.name}' is not yet empty. waiting...")
-            sleep(sleep_interval)
-    except Exception as e:
-        raise
-    finally:
-        # ensure that the lifecycle rule is removed
-        bucket.lifecycle_rules = []
-        bucket.patch()
+    lifecycle_rules = list(bucket.lifecycle_rules)
+
+    prune_lifecycle_already_exists: bool = False
+    for lifecycle in lifecycle_rules:
+        if is_prune_lifecycle(lifecycle):
+            lifecycle["condition"]["createdBefore"] = prune_rule["condition"][
+                "createdBefore"
+            ]
+            prune_lifecycle_already_exists = True
+            break
+
+    if not prune_lifecycle_already_exists:
```

Comment:
Here, this is the perfect candidate for a `for ...: else:` loop:

```python
for lifecycle in lifecycle_rules:
    if is_prune_lifecycle(lifecycle):
        ... # rest of the code
        break
else:
    lifecycle_rules.append(prune_rule)
```
Also, it's better in this case to avoid the creation of new lists. We have `append` and `extend` to mutate lists, they are not immutable and using a `+` implies the creation here of two new lists, one for `[prune_rule]` and the final one while being less explicit than using `append`

Reply:
Didn't knew about the else condition, very cool!
Yeh, I thought about the list thing at time of writing the code, but I ended up doing that to keep mypy happy :')